src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -1,214 +1,144 @@
+
+    tabbedPane.addChangeListener {
+      if (tabbedPane.selectedIndex == CHAT_TAB_INDEX) {
+        val currentChatPanel = chatContainerPanel.getComponent(0) as? ChatPanel
+        currentChatPanel?.requestFocusOnInputPrompt()
+      }
+    }
 
-    // Main content panel
-    contentPanel.layout = BorderLayout(0, 0)
-    contentPanel.border = BorderFactory.createEmptyBorder(0, 0, 10, 0)
-    contentPanel.add(chatPanel, BorderLayout.CENTER)
-    contentPanel.add(lowerPanel, BorderLayout.SOUTH)
-    tabbedPane.addChangeListener { focusPromptInput() }
-    val singInWithSourcegraphPanel = SignInWithSourcegraphPanel(project)
     allContentPanel.add(tabbedPane, "tabbedPane", CHAT_PANEL_INDEX)
-    allContentPanel.add(
-        singInWithSourcegraphPanel, SING_IN_WITH_SOURCEGRAPH_PANEL, SIGN_IN_PANEL_INDEX)
-    allContentLayout.show(allContentPanel, SING_IN_WITH_SOURCEGRAPH_PANEL)
-    refreshPanelsVisibility()
+    allContentPanel.add(signInWithSourcegraphPanel, SIGN_IN_PANEL, SIGN_IN_PANEL_INDEX)
+    allContentLayout.show(allContentPanel, SIGN_IN_PANEL)
 
+    ApplicationManager.getApplication().invokeLater { refreshPanelsVisibility() }
     ApplicationManager.getApplication().executeOnPooledThread { refreshSubscriptionTab() }
-    refreshChatToEmpty()
-    historyTree.refreshTree()
+
+    startNewChat(project)
   }
 
   @RequiresBackgroundThread
   fun refreshSubscriptionTab() {
     CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
-      fetchSubscriptionPanelData(project, agent.server).thenAccept {
-        if (it != null) {
-
-          ApplicationManager.getApplication().invokeLater { refreshSubscriptionTab(it) }
+      val data = fetchSubscriptionPanelData(project, agent.server)
+      if (data != null) {
+        val isSubscriptionTabPresent = tabbedPane.tabCount >= SUBSCRIPTION_TAB_INDEX + 1
+        if (data.isDotcomAccount && data.codyProFeatureFlag) {
+          if (!isSubscriptionTabPresent) {
+            tabbedPane.insertTab(
+                /* title = */ "Subscription",
+                /* icon = */ null,
+                /* component = */ subscriptionPanel,
+                /* tip = */ null,
+                SUBSCRIPTION_TAB_INDEX)
+          }
+          subscriptionPanel.update(data.isCurrentUserPro)
+        } else if (isSubscriptionTabPresent) {
+          tabbedPane.remove(SUBSCRIPTION_TAB_INDEX)
         }
       }
     }
   }
 
-  @RequiresEdt
-  private fun refreshSubscriptionTab(data: SubscriptionTabPanelData) {
-    val isSubscriptionTabPresent = tabbedPane.tabCount >= SUBSCRIPTION_TAB_INDEX + 1
-
-    if (data.isDotcomAccount && data.codyProFeatureFlag) {
-      if (!isSubscriptionTabPresent) {
-        tabbedPane.insertTab(
-            /* title = */ "Subscription",
-            /* icon = */ null,
-            /* component = */ subscriptionPanel,
-            /* tip = */ null,
-            SUBSCRIPTION_TAB_INDEX)
+  private fun runNewCommand(project: Project, commandId: CommandId) {
```

Comment:
`project` is a private val of this class already - there is no need to pass it here